packages/devtools_app/lib/src/shared/table/table_data.dart

```diff
@@ -13,7 +13,17 @@ import '../theme.dart';
 import '../ui/colors.dart';
 import '../utils.dart';
 
-// TODO(peterdjlee): Remove get from method names.
+/// Defines how a column should display data in a table.
+///
+/// [ColumnData] objects should be defined as static where possible, and should
+/// not manage any stateful data. The table controllers expect columns to be
+/// solely responsible for declaring how to layout table data.
+///
+/// Any data that can't be stored on the [dataObject] may be accessed by passing
+/// a long-living controller to the constructor of the [ColumnData] subclass.
+///
+/// The controller is expected to be alive for the duration of the app
+/// connection.
```

Comment:
What happens after reconnection? Does table reset and start using new ColumnData instances?

Reply:
good question. I added dispose method to the table state classes to ensure we null out the table controller, which would mean it is guaranteed to be recreated for a new widget if the state stays alive. This should never happen, but it doesn't hurt to be extra cautious. 

If a column is not static (because a controller has to be passed to it), then it will be recreated for a new instance of the widget that creates the column with whatever controller is passed. 

When DevTools disconnects from an app, the existing controllers are destroyed and new ones will be created for the next app connection.